google/cloud/forseti/common/gcp_type/firewall_rule.py

```diff
@@ -350,8 +349,9 @@ def _validate_direction(self):
               * _destination_ranges is set
         """
         if self.direction == 'INGRESS':
-            if (not self._source_ranges and not self._source_tags and not
-                    self.source_service_accounts):
+            if (not self._source_ranges and
+                    not self._source_tags and
+                    not self.source_service_accounts):
                 raise InvalidFirewallRuleError(
                     'Ingress rule missing required field oneof "sourceRanges" '
                     'or "sourceTags" or "sourceServiceAccounts": "%s".' % self)
```

Comment:
I am really not good with compound conditions.  But based on the message of the raised exception, should the operator inside the `if` before `or` instead of `and`?

Reply:
The logic here hasn't changed if that's what you're asking just following pep8 guideline:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Reply:
Okay, I filed an issue here for mark to take a look.